components/popover/index.js

```diff
@@ -321,7 +322,10 @@ class Popover extends Component {
 			content = <Fill name={ SLOT_NAME }>{ content }</Fill>;
 		}
 
-		return <span ref={ this.bindNode( 'anchor' ) }>{ content }</span>;
+		return <span ref={ this.bindNode( 'anchor' ) }>
+			{ content }
+			{ this.state.isMobile && expandOnMobile && <MobileScrollLock /> }
```

Comment:
This component would benefit from using this new HOC `withViewportMatch`, too.

It's still tricky because it depends on another prop. I think we could add additional checks to HOC:
```js
export default withMobileScrollLock(
    ( props ) => Boolean( props.expandOnMobile ),
)( Popover );
```